test/test_manifest.json

```diff
@@ -3659,6 +3659,16 @@
       "type": "eq",
       "about": "CFF font that is drawn with clipping."
     },
+    {  "id": "prefilled_f1040",
+       "file": "pdfs/prefilled_f1040.pdf",
+       "md5": "6c45b2c43e365bb7c88b9cddca0a4865",
+       "rounds": 1,
+       "type": "eq",
+       "print": true,
+       "annotationStorage": {
+         "1605R": true
```

Comment:
I verified that this patch itserlf fixes the reported issue, but I can't get this reference test working when checking out this patch and I also don't see how this is supposed to work. The prefilled file seems to only have prefilled content on page 1 and this page is always rendered, so there is an annotation layer and the content is simply in the annotation storage. Checkbox 1605R is also on the first page, so that will also work. I don't see any changes in the output before/after this patch, which confirms my suspicion that this most likely doesn't work as expected.

Did you try the suggestion from https://github.com/mozilla/pdf.js/pull/12395#issuecomment-695767556? You can run the reference tests locally before/after applying this patch to check if the test indeed works as expected.